specification/desktopvirtualization/resource-manager/Microsoft.DesktopVirtualization/preview/2021-05-13-preview/desktopvirtualization.json

```diff
@@ -3637,181 +3966,496 @@
+        "joinType": {
+          "enum": [
+            "ActiveDirectory",
+            "AzureActiveDirectory"
+          ],
+          "x-ms-enum": {
+            "name": "DomainJoinType",
+            "modelAsString": true,
+            "values": [
+              {
+                "value": "ActiveDirectory",
+                "description": "Using microsoft active directory."
+              },
+              {
+                "value": "AzureActiveDirectory",
+                "description": "Using microsoft azure active directory."
+              }
+            ]
+          },
+          "type": "string",
+          "description": "The type of domain join done by the virtual machine.",
+          "x-nullable": false
+        },
+        "mdmProviderGuid": {
+          "type": "string",
+          "description": "The MDM Provider GUID used during MDM enrollment for Azure AD joined virtual machines."
+        },
+        "credentials": {
+          "$ref": "#/definitions/CredentialsProperties",
+          "description": "Credentials needed to create the virtual machine."
+        }
+      }
+    },
+    "CredentialsProperties": {
+      "description": "Credentials needed to create the virtual machine.",
+      "type": "object",
+      "properties": {
+        "localAdmin": {
+          "$ref": "#/definitions/KeyvaultCredentialProperties",
+          "description": "The local admin credentials."
+        },
+        "domainAdmin": {
+          "$ref": "#/definitions/KeyvaultCredentialProperties",
+          "description": "The domain admin credentials."
+        }
+      }
+    },
+    "KeyvaultCredentialProperties": {
+      "description": "Credentials kept in the keyvault.",
+      "type": "object",
+      "properties": {
+        "userName": {
+          "type": "string",
+          "description": "The user name to the account."
+        },
+        "passwordKeyVaultResourceId": {
+          "type": "string",
+          "description": "The keyvault resource id to the keyvault secrets."
+        },
+        "passwordSecretName": {
+          "type": "string",
+          "description": "The keyvault secret name the password is stored in."
+        }
+      }
+    },
+    "HostPoolUpdateParameters": {
+      "description": "Represents properties for a hostpool update.",
+      "type": "object",
+      "properties": {
+        "saveOriginalDisk": {
+          "type": "boolean",
+          "description": "Whether to save original disk. False by default."
+        },
+        "maxVMsRemovedDuringUpdate": {
+          "type": "integer",
+          "format": "int32",
+          "description": "The maximum virtual machines to be removed during hostpool update."
+        },
+        "maintenanceAlerts": {
+          "type": "array",
+          "description": "The alerts given to customers for hostpool update.",
+          "items": {
+            "$ref": "#/definitions/MaintenanceAlertsProperties"
+          }
+        },
+        "logOffDelaySeconds": {
```

Comment:
I would like to ask log off message is PC message, which is a message popped up on VM screen, right? if so, should we have a maintenaceMessages property as array to define both logoff message and cancel update message.  Also I think we should group logOffDelaySeconds and logOffMessage together.

Reply:
@shailahirji

Reply:
not following what cancel update message means here "logoff message and cancel update message". By grouping if you mean create an object that contains logOffDelaySeconds  and logOffMessage as properties -sure

Reply:
Sheikh suggested during last meeting that when customer cancel host pool update, we would like to support to send cancel host pool update meeting like logoff message.

> By grouping if you mean create an object that contains logOffDelaySeconds and logOffMessage as properties -sure
yes.

Reply:
My suggestion: 
{
   "sessionHostMessages": [
      { "type": "logOff", "timeout": "30", "message": "xxxxxxx" }
      { "type": "cancelUpdate", "timeout": "30", "message": "xxxxxxx" }
   ]
}

Reference: I found the wording "Timeout" from https://docs.microsoft.com/en-us/windows-server/administration/windows-commands/shutdown